src/serdes/binary-serializer.ts

```diff
@@ -1,110 +1,121 @@
-import { strict as assert } from "assert";
-import { parseBytes, bytesToHex } from "../utils/bytes-utils";
-import { makeClass } from "../utils/make-class";
-import { Field } from "../definitions";
-
-const BytesSink = {
-  put(/* bytesSequence */) {
-    // any hex string or any object with a `length` and where 0 <= [ix] <= 255
-  },
-};
-
-const BytesList = makeClass(
-  {
-    implementing: BytesSink,
-    BytesList() {
-      this.arrays = [];
-      this.length = 0;
-    },
-    put(bytesArg) {
-      const bytes = parseBytes(bytesArg, Uint8Array);
-      this.length += bytes.length;
-      this.arrays.push(bytes);
-      return this;
-    },
-    toBytesSink(sink) {
-      this.arrays.forEach((arr) => {
-        sink.put(arr);
-      });
-    },
-    toBytes() {
-      const concatenated = new Uint8Array(this.length);
-      let pointer = 0;
-      this.arrays.forEach((arr) => {
-        concatenated.set(arr, pointer);
-        pointer += arr.length;
-      });
-      return concatenated;
-    },
-    toHex() {
-      return bytesToHex(this.toBytes());
-    },
-  },
-  undefined
-);
-
-const BinarySerializer = makeClass(
-  {
-    BinarySerializer(sink) {
-      this.sink = sink;
-    },
-    write(value) {
+import * as assert from "assert";
+import { Field, FieldInstance } from "../definitions";
+
+/*
+ * @brief: Bytes list is a collection of buffer objects, with length denoting the totaly number of bytes
+ */ 
+class BytesList {
+  arrays: Array<Buffer> = [];
+  length = 0;
```

Comment:
Rather than keeping this as a property that needs to be updated you could just make it as a getter:
```
public getLength() {
  return this.bytes.reduce(0, (accumulated, next) => {
    return accumulated + next.length
  }
}

```

This simplifies your code because you are keeping track of less state.

Reply:
Done